lib_dsp/api/dsp_testing.h

```diff
@@ -62,127 +70,141 @@ typedef enum {
-
-dsp_complex_int16_t dsp_conv_complex_float_to_complex_int16(dsp_complex_float_t x, const int x_exp, int *error);
-dsp_complex_int32_t dsp_conv_complex_float_to_complex_int32(dsp_complex_float_t x, const int x_exp, int *error);
-
-dsp_ch_pair_int16_t dsp_conv_ch_pair_float_to_ch_pair_int16(dsp_ch_pair_float_t x, const int x_exp, int *error);
-dsp_ch_pair_int32_t dsp_conv_ch_pair_float_to_ch_pair_int32(dsp_ch_pair_float_t x, const int x_exp, int *error);
+dsp_float_t dsp_conv_int8_to_float (const int8_t x,  const int x_exp, int * UNSAFE error);
+dsp_float_t dsp_conv_uint8_to_float(const uint8_t x, const int x_exp, int * UNSAFE error);
+dsp_float_t dsp_conv_int16_to_float (const int16_t x,  const int x_exp, int * UNSAFE error);
+dsp_float_t dsp_conv_uint16_to_float(const uint16_t x, const int x_exp, int * UNSAFE error);
+dsp_float_t dsp_conv_int32_to_float (const int32_t x,  const int x_exp, int * UNSAFE error);
+dsp_float_t dsp_conv_uint32_to_float(const uint32_t x, const int x_exp, int * UNSAFE error);
+dsp_float_t dsp_conv_int64_to_float (const int64_t x,  const int x_exp, int * UNSAFE error);
+dsp_float_t dsp_conv_uint64_to_float(const uint64_t x, const int x_exp, int * UNSAFE error);
+
+int8_t  dsp_conv_float_to_int8 (dsp_float_t d, const int d_exp, int * UNSAFE error);
+uint8_t dsp_conv_float_to_uint8(dsp_float_t d, const int d_exp, int * UNSAFE error);
+int16_t  dsp_conv_float_to_int16 (dsp_float_t d, const int d_exp, int * UNSAFE error);
+uint16_t dsp_conv_float_to_uint16(dsp_float_t d, const int d_exp, int * UNSAFE error);
+int32_t  dsp_conv_float_to_int32 (dsp_float_t d, const int d_exp, int * UNSAFE error);
+uint32_t dsp_conv_float_to_uint32(dsp_float_t d, const int d_exp, int * UNSAFE error);
+int64_t  dsp_conv_float_to_int64 (dsp_float_t d, const int d_exp, int * UNSAFE error);
+uint64_t dsp_conv_float_to_uint64(dsp_float_t d, const int d_exp, int * UNSAFE error);
+
+dsp_complex_float_t dsp_conv_complex_int16_to_complex_float(dsp_complex_int16_t x, const int x_exp, int * UNSAFE error);
+dsp_complex_float_t dsp_conv_complex_int32_to_complex_float(dsp_complex_int32_t x, const int x_exp, int * UNSAFE error);
+
+dsp_ch_pair_float_t dsp_conv_ch_pair_int16_to_ch_pair_float(dsp_ch_pair_int16_t x, const int x_exp, int * UNSAFE error);
+dsp_ch_pair_float_t dsp_conv_ch_pair_int32_to_ch_pair_float(dsp_ch_pair_int32_t x, const int x_exp, int * UNSAFE error);
+
+dsp_complex_int16_t dsp_conv_complex_float_to_complex_int16(dsp_complex_float_t x, const int x_exp, int * UNSAFE error);
+dsp_complex_int32_t dsp_conv_complex_float_to_complex_int32(dsp_complex_float_t x, const int x_exp, int * UNSAFE error);
+
+dsp_ch_pair_int16_t dsp_conv_ch_pair_float_to_ch_pair_int16(dsp_ch_pair_float_t x, const int x_exp, int * UNSAFE error);
+dsp_ch_pair_int32_t dsp_conv_ch_pair_float_to_ch_pair_int32(dsp_ch_pair_float_t x, const int x_exp, int * UNSAFE error);
```

Comment:
These are either untested/unused, or tested as part of testing the `conv_vect` functions.